### PR TITLE
Pytest: Added minor fixes to unittest

### PR DESCRIPTION
**Pull request by mitzkia**
- eliminate duplicated variables and typos
- fix multiple registering of temporary files
- remove closing of PosixPath type of files

Signed-off-by: Andras Mitzki <andras.mitzki@balabit.com>

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: tests/pytest_framework/src/setup/unit_testcase.py [hunk 1]
```diff
@@ -79,22 +78,18 @@ def prepare_input_file(self, input_content):
         input_file_path = self.get_temp_file()
         writeable_file = open_file(input_file_path, "a+")
         readable_file = open_file(input_file_path, "r")
-        self.__registered_files.append(writeable_file)
```
[the remark this answers is not in the document]
Comment (mitzkia) [reply tests/pytest_framework/src/setup/unit_testcase.py]: @gaborznagy  Thanks for pointing on this issue.  It was a confusion between file paths and file fds, my original goal was to register both created file paths and opened file fds, and on teardown make a proper delete of this. But I made some typos, and it causes a little confusion.

File: tests/pytest_framework/src/setup/unit_testcase.py [hunk 2]
```diff
@@ -79,22 +78,18 @@ def prepare_input_file(self, input_content):
...
         return writeable_file, readable_file
 
     def __teardown(self):
-        for registered_file in self.__registered_files:
-            registered_file.close()
-
-        for temp_file in self.__registerd_files:
+        for temp_file in self.__registered_files:
             if temp_file.exists():
                 temp_file.unlink()
```
[the remark this answers is not in the document]
Comment (mitzkia) [reply tests/pytest_framework/src/setup/unit_testcase.py]: Thanks for this note also. With the new update I first close the opened fds, and after that I can safely delete the files.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**